### DISCUSSION TITLE
m2o, o2m and m2m: select fields from the current record

### DISCUSSION THREAD

**Discussion by aliemba** [directus/directus#10612] [Feature Requests]
It would be great if you could select fields from the current record. Lets say you have a page that is related to a country. I would like to add a page category to the page - from a collection with page categories pr. country - and only show the users page categories, that are from the pages country.

So; i would then have the feature to only select page categories that is related to the specific pages country.

This relates to https://github.com/directus/directus/pull/9135

**Comment by azrikahar** [comment]
I believe this should technically already be possible using Conditional Fields, but there seems to be an issue with relations in the conditions configuration 🤔 Did a quick patch around these parts just for testing: https://github.com/directus/directus/blob/4ba3ef4070b543dacdb25b7d3b3949b8ef77dcce/app/src/interfaces/_system/system-interface-options/system-interface-options.vue#L91-L120

And it seems to have worked:

https://user-images.githubusercontent.com/42867097/146721761-608e8cb4-315c-4e33-a29e-fddffa0208c7.mp4

Hopefully this is what you are looking for? but there should be a better fix here that I'm not aware of.

**Reply by samthompsonkennedy** [reply]
I think this is what I'm trying to do at the moment, what was the syntax you used for this? Maybe if you could shoot through a screenshot of what you did?

Thanks!

**Reply by azrikahar** [reply]
Ideally we can configure this via the [Conditions  tab](https://docs.directus.io/configuration/data-model/#conditions) when you're configuring the field (which was what I did in the video), but unfortunately I had to hack the code block above as I mentioned to make it work. 

There's likely a more complete solution tbh, though I'm not actively looking into this as of now.

**Reply by neo7-studio-web** [reply]
That would be awesome ! Just curious : why in the condition tab rather than the filter option ?

**Reply by azrikahar** [reply]
We are indeed using filter option to limit the records on the dropdown, but conditional fields will be the one that allow us to control what is the value of said filter.

Another approach would be to allow filter to use something like dynamic variable to refer the other field (which might be what you are thinking of), but that's not something built-in at the moment, hence the suggestion of doing it via conditional fields. 👍

**Comment by abdonrd** [comment]
That would be great!

In my use case, I have:

- **Profile**
· name
- **Match**
· participants (M2M Profile)
· winners (M2M Profile)

And I want to filter the `match.winners` interface to only allow profiles that are part of `match.participants`.

**Reply by abdonrd** [reply]
I just updated my comment with a better example.

**Reply by abdonrd** [reply]
@azrikahar is this currently possible with something you've said about the `Interface > Filter` or `Conditions`? Thanks!

**Reply by azrikahar** [reply]
It is not possible _now_ per se, but yea it could be achievable once this is made possible in the sense that the UI will show limited selections based on `match.participants`. However if you're more so thinking of validation of the field value itself, then that would fall under Field Validation that is discussed over here: https://github.com/directus/directus/discussions/4390. Hope that clear things up!

**Reply by abdonrd** [reply]
Understood! Thanks for the info!

For now I was just thinking of showing the limited selections based on `match.participants`, but if it worked as well as validation it would be ideal!

**Comment by aliemba** [comment]
I understand that this could be done (or be done in in the future) via conditions.

But the best solution would be if it sits on the `Filter` when configuring the relation. Instead of filtering for specific values - filtering based upon filed on the current collection.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
